packages/core/utils/src/dml/entity.ts

```diff
@@ -16,8 +19,34 @@ import { BelongsTo } from "./relations/belongs-to"
 export class DmlEntity<Schema extends DMLSchema> implements IDmlEntity<Schema> {
   [IsDmlEntity]: true = true
 
+  name: string
+
+  readonly #tableName: string
   #cascades: EntityCascades<string[]> = {}
-  constructor(public name: string, public schema: Schema) {}
+
+  constructor(nameOrConfig: Config, public schema: Schema) {
+    if (isString(nameOrConfig)) {
+      const [schema, ...rest] = nameOrConfig.split(".")
+      const name = rest.length ? rest.join(".") : schema
+      this.name = toCamelCase(name)
```

Comment:
Let's rename `name` to `modelName` everywhere so it's clear what it is (I wasn't sure what it represents until I saw it below in the code)

Reply:
I wont do this one because here it is the name of the object, not the model name. there is no notion of model until we reach the mikro orm helper and here we use the name to define the model name. Same thing as the name of a function or a class that we emulate here `Object.name` basically 😄

Reply:
But isn't this the DML model name? eg. in `const user = model.define("user", ...` `"user"` would be the name of the model (thus, the function being called `model.define`?

Reply:
no it is not, same as when you passed the entity to the MedusaService, we were using the Class.name to infer everything, here it is the exact same behavior but instead of the name of the class (cause you can still pass mikro orm class or other classes if you want) we are using the DML.name

Reply:
Got it 👍